contracts/src/v0.8/functions/v1_1_0/libraries/L1Block.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Copied from the implementation contract for https://optimistic.etherscan.io/address/0x4200000000000000000000000000000000000015#readProxyContract
+// https://optimistic.etherscan.io/address/0x07dbe8500fc591d1852b76fee44d5a05e13097ff#code
+
+import { ISemver } from "./ISemver.sol";
+
+/// @custom:proxied
+/// @custom:predeploy 0x4200000000000000000000000000000000000015
+/// @title L1Block
+/// @notice The L1Block predeploy gives users access to information about the last known L1 block.
+///         Values within this contract are updated once per epoch (every L1 block) and can only be
+///         set by the "depositor" account, a special system address. Depositor account transactions
+///         are created by the protocol whenever we move to a new epoch.
+contract L1Block is ISemver {
+    /// @notice Address of the special depositor account.
+    address public constant DEPOSITOR_ACCOUNT = 0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0001;
+
+    /// @notice The latest L1 block number known by the L2 system.
+    uint64 public number;
+
+    /// @notice The latest L1 timestamp known by the L2 system.
+    uint64 public timestamp;
+
+    /// @notice The latest L1 base fee.
+    uint256 public basefee;
+
+    /// @notice The latest L1 blockhash.
+    bytes32 public hash;
+
+    /// @notice The number of L2 blocks in the same epoch.
+    uint64 public sequenceNumber;
+
+    /// @notice The scalar value applied to the L1 blob base fee portion of the blob-capable L1 cost func.
+    uint32 public blobBaseFeeScalar;
+
+    /// @notice The scalar value applied to the L1 base fee portion of the blob-capable L1 cost func.
+    uint32 public baseFeeScalar;
+
+    /// @notice The versioned hash to authenticate the batcher by.
+    bytes32 public batcherHash;
+
+    /// @notice The overhead value applied to the L1 portion of the transaction fee.
+    /// @custom:legacy
+    uint256 public l1FeeOverhead;
+
+    /// @notice The scalar value applied to the L1 portion of the transaction fee.
+    /// @custom:legacy
+    uint256 public l1FeeScalar;
+
+    /// @notice The latest L1 blob base fee.
+    uint256 public blobBaseFee;
+
+    /// @custom:semver 1.2.0
+    string public constant version = "1.2.0";
```

Comment:
If we want to do something like this, let's re-use the Chainlink style pattern using `ITypeAndVersion`
```
string public constant override typeAndVersion = "L1Block 1.0.0";
```